op-batcher/batcher/driver.go

```diff
@@ -550,9 +567,39 @@ func (l *BatchSubmitter) cancelBlockingTx(queue *txmgr.Queue[txRef], receiptsCh
 
 // sendTransaction creates & queues for sending a transaction to the batch inbox address with the given `txData`.
 // The method will block if the queue's MaxPendingTransactions is exceeded.
-func (l *BatchSubmitter) sendTransaction(ctx context.Context, txdata txData, queue *txmgr.Queue[txRef], receiptsCh chan txmgr.TxReceipt[txRef]) error {
+func (l *BatchSubmitter) sendTransaction(ctx context.Context, txdata txData, queue *txmgr.Queue[txRef], receiptsCh chan txmgr.TxReceipt[txRef], daGroup *errgroup.Group) error {
 	var err error
-	// Do the gas estimation offline. A value of 0 will cause the [txmgr] to estimate the gas limit.
+
+	// if Alt DA is enabled we post the txdata to the DA Provider and replace it with the commitment.
+	if l.Config.UseAltDA {
+		// sanity checks
+		if nf := len(txdata.frames); nf != 1 {
+			l.Log.Crit("Unexpected number of frames in calldata tx", "num_frames", nf)
+		}
+		if txdata.asBlob {
+			l.Log.Crit("Unexpected blob txdata with AltDA enabled")
+		}
+
+		// when posting txdata to an external DA Provider, we use a goroutine to avoid blocking the main loop
+		// since it may take a while for the request to return.
+		// the number of concurrent goroutines is limited by the semaphore check (daGroup.TryAcquire(1)) in publishTxToL1
+		daGroup.Go(func() error {
```

Comment:
At a first glance, this looks like it might block due to the errgroup limit. A goroutine could be started by a different process between the `tryGo` in `publishTxToL1` and here.

But seeing the whole program, we know that both are only called in the main loop, so we can actually do this safely. And it is simpler than the concurrency-safe approach.

Reply:
The concurrency-safe approach being to use TryGo again here and put back the frames if didn’t start? Honestly the more I think about it the more I don’t dislike that approach. Also means we can get rid of the previous tryGo and we make the logic more localized, as you are explaining. Probably leads to less confusion?

Reply:
Thanks for that PR! Incorporated it with some other small modifs about logging that we talked about during our sync: [077204d](https://github.com/Layr-Labs/optimism/pull/7/commits/077204d7acb084b218881f34b74d32f61a5b4558)